### PR TITLE
feat: Use Caddy server

### PR DESCRIPTION
**Pull request by dunglas**
* Production, development and CI ready
* Automatic HTTPS (in dev and in prod!)
* HTTP/2, HTTP/3 and [Server Push](https://symfony.com/doc/current/web_link.html) support
* [Vulcain](https://vulcain.rocks)-enabled
* Just 2 services (PHP FPM and Caddy server)
* Super-readable configuration

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -17,11 +17,21 @@ services:
...
+      SERVER_NAME: ${SERVER_NAME:-localhost}
+    restart: unless-stopped
     ports:
       - "80:80"
+      - "443:443"
+    volumes:
+      - caddy_data:/data
+      - caddy_config:/config
+
+
```
Comment (chalasr) [inline docker-compose.yml:34]: extra blank line

**Review by chalasr — APPROVED** [review]
👍

File: Dockerfile [hunk 2]
```diff
@@ -109,56 +109,15 @@ RUN chmod +x /usr/local/bin/docker-entrypoint
 ENTRYPOINT ["docker-entrypoint"]
 CMD ["php-fpm"]
 
+FROM caddy:${CADDY_VERSION}-builder-alpine AS symfony_caddy_builder
 
-# "nginx" stage
-# depends on the "php" stage above
-FROM nginx:${NGINX_VERSION}-alpine AS symfony_nginx
+RUN xcaddy build \
+    --with github.com/dunglas/vulcain/caddy
```
Comment (maxhelias) [inline Dockerfile:115]: github.com/dunglas/vulcain/tree/master/caddy ? Missing `tree/master`

**Review by maxhelias — APPROVED** [review]
It's a great move! I like the idea with Caddy :)

File: Dockerfile (hunk 2 above)
Comment (dunglas) [reply Dockerfile:115]: No it's the name of the Go module, it's intended.

**Pull request merged**